### ISSUE TITLE
[7.5+] Transform applied to image speeding up during a load [minor]

### ISSUE THREAD

**Issue by Org-gith** [renpy/renpy#3706]
[Tested it on an older prerelease of 7.5 and the newest 7.5 release and the issue happens, this doesn't occur on 7.4.11 and prior.]

This is somewhat minor but I have started noticing that the transforms applied to an image are getting 'speeded up' under certain situations, this occurs during what I assume is a loading of the images into the cache.

Here is a video to showcase the weird speed-up that occurs, here it is showcased on the main menu (left is 7.4.11 doing fine, right is 7.5 doing the issue where every transform is speeding up for a short instant at the beginning on the main menu screen):

https://user-images.githubusercontent.com/78823063/175849778-0034cca6-30b6-44ab-912c-19f99d377434.mp4

(The easier to really witness the issue is to focus on the big red arrow at the bottom right, you will see it clearly getting sped up on 7.5 when I get to the main menu screen. The rest of the main menu's images too but it might be less noticeable for someone that has not worked on this project for a while.
The menu is made of images which have transforms applied to them, warping things like easein / easeout / linear / alpha, nothing is prerecorded or a video.
This also happens by launching the game for the first time and getting into the main menu, but it was easier to record it that way by loading a save and then going back to the main menu repetitively.)
_____________________________________________________________________________________________________________________________________

Edit: tested up to the 26th May nightly SDK in an attempt to identify the commit that made this occur, sadly all the nightly versions available right now have the issue so wasn't able to find the first time when it occurred

**Comment by Org-gith** [comment]
Okay after some research, it seems to be tied to start_predict or predicting. 
When I remove this code from the screen main_menu():
```
on "show":
   action With(dissolve), Play("music", "audio/bgm/main menu - track.ogg")
```
The weird speed-up doesn't occur when I get to the main menu without this code, but what happens instead is simply a lag lasting between 1 and 2 seconds (the game getting a hiccup). So the acceleration seems only a symptom of With(dissolve) trying to catch up with the lag of loading the images of the main menu screen.
But all that doesn't happen in 7.4.11 and I think there might be a change in image prediction that might cause this for the main menu. I predict all the animals used for the main_menu and more during the presplash screen, the prediction seems to be working since some images inside the game and not for the main menu that is usually slow to show up are instant but it feels like it's not working anymore for the main_menu or maybe it actually runs only when it reaches the main_menu when it's too late. 
I tried to remove or keep the start predict and performance doesn't seem to change for the main_menu but for the rest, it seems to work.

This is a somewhat a minor issue since it seems to affect only the main menu somehow, and can be patched in my case by adding a black screen in front of the main menu for 1-2 sec to make things load properly so that the user doesn't feel the hiccup, but this is also because I use a good number of anims that would need to be predicted to not do the hiccup but it's apparently not getting predicted anymore so it's a bit niche.

**Comment by renpytom** [comment]
Do you have the original version, with the problem? I want to just make sure we're not having some sort of issue where the prediction thread is blocking the main thread.

**Comment by Org-gith** [comment]
I can get that ready by tomorrow and send you through email (1 am over here), just to be sure the version you want to see is the version that has the transform speed-up the problem even with the new pause predict, or the one before that had prediction with a standard pause and also had the transform speed up problem? Both make the transform speed up at arrival.

**Comment by renpytom** [comment]
I'll be honest that I don't completely understand what's going wrong. Would it be possible to have something I can switch back and forth between both versions?

**Comment by Org-gith** [comment]
I really don't get it either, I think the "important" part is that the action With(dissolve) creates an acceleration when showing the main menu which didn't happen in the past, so I'll focus on that but I'll include a button when launching the project, that either disable or enable the new preload (which is a great improvement with the ability to pause until it's complete), and another one to disable/enable the action With(dissolve) on the main menu. Sending that through email.

**Comment by Org-gith** [comment]
Okay, got a really strange discovery this morning, I was able to fix the issue randomly by putting default setting on Nvidia's control panel  then ended up launching the game and it had no issues. Then I made some investigation, and I found the exact setting for Nvidia that creates this issue, it happens when I disable Vsync on the nvidia's control panel. I always disable Vsync on a GPU control panel since I have an Adaptive Sync monitor for some years now, so it's usually better to always have the monitor run without Vsync.
Since 7.5 for ren'py if you don't have Vsync enabled in your GPU control panel it will do this acceleration thing (I had it turned off for years, and prior to 7.4.11 it wouldn't affect a game like this). Overall it's a very minor issue but in case that's something, otherwise, we can definitely close since it seems very niche.

**Comment by renpytom** [comment]
For a visual novel Vsync makes sense - you don't want to run faster than the monitor can display. I can't repeat this here with Vsync enabled, so I'm going to close this issue. This really doesn't seem like something major at this point. 

Ren'Py has been using adaptive sync when the hardware supports it, by providing a negative value to SDL_GL_SetSwapInterval.

**Comment by Org-gith** [comment]
Totally agree, it's very minor and niche and in fact... 

Not an issue, but just wanted to share some more discoveries (maybe you are aware but in case that's something...)
preferences.gl_tearing = True doesn't actually remove Vsync, in fact after some testing/monitoring, it is reliably less performant than preferences.gl_tearing = False by a very negligible ms/frame. But it does prevent that weird niche issue I had to have it on true, but Vsync is definitely on in both cases. Preferences.gl_framerate doesn't work. I can't explain it.
I have tested removing gl2draw.pyx and gldraw.pyx from the ren'py SDK folder and relaunching the launcher to see what happens, the game still behave exactly the same after removal, it seems something is taking priority over the actual .py.


However, setting directly in the project:
```
init python hide:
   import os
   os.environ["RENPY_GL_VSYNC"] = "0"
```

This removes Vsync entirely and I'm getting fps in the 200s whereas before I would be locked at my refresh rate (75hz) no matter what I tried. Preferences.gl_framerate still doesn't work though so it's really not something that you would run without a sort of frame cap anyway and I think I might stay preferences.gl_tearing = True which keeps Vsync ON but removes the strange issue when I set some incompatible settings in Nvidia panel.
